pkg/lib/packagekit.js

```diff
@@ -87,13 +87,19 @@ export function call(objectPath, iface, method, args, opts) {
 }
 
 /**
- * Figure out whether PackageKit is available
+ * Figure out whether PackageKit is available and useable
  */
 export function detect() {
-    return call("/org/freedesktop/PackageKit", "org.freedesktop.DBus.Properties",
-                "Get", [ "org.freedesktop.PackageKit", "VersionMajor" ])
-            .then(() => true,
-                  () => false);
+    return cockpit.spawn([ "findmnt", "-T", "/usr", "-n", "-o", "VFS-OPTIONS" ])
+            .then(options => {
+                if (options.split(",").indexOf("ro") >= 0)
+                    return false;
+                else
+                    return call("/org/freedesktop/PackageKit", "org.freedesktop.DBus.Properties",
+                                "Get", [ "org.freedesktop.PackageKit", "VersionMajor" ])
+                            .then(() => true,
+                                  () => false);
+            });
```

Comment:
Although the findmnnt Should Not Fail™, let's please add a catch() handler with a log at least.

Reply:
I made it so that we fall back to checking the D-Bus API when findmnt fails.

Reply:
Good catch!